### PR TITLE
MCOL-537 Address existing compiler warnings.

### PR DESCRIPTION
**Pull request by drrtuy**

### CODE REVIEW

File: ddlproc/ddlproc.cpp
Comment (pleblanc1976) [inline ddlproc/ddlproc.cpp]: Since we're handling the error now, and it's likely important that this succeed, this should exit after failure, IMO.

File: oamapps/alarmmanager/alarmmanager.cpp
Comment (pleblanc1976) [inline oamapps/alarmmanager/alarmmanager.cpp]: looks like we lost indentation here

File: primitives/linux-port/column.cpp
Comment (pleblanc1976) [inline primitives/linux-port/column.cpp:1424]: I'm surprised casting is necessary in all of these memcpy calls.  Do you know which version of gcc is complaining about this?

File: tools/dbloadxml/colxml.cpp
Comment (pleblanc1976) [inline tools/dbloadxml/colxml.cpp]: My guess is that failure there isn't always a real problem.  Failures later would tell the user whether it was a real problem or not.  But also this must be ancient code written under different specs.

**Review by pleblanc1976 — APPROVED** [review]

**Review by pleblanc1976 — COMMENTED** [review]
It's probably fine, but we should make sure it still compiles on our supported platforms before merging.

**Comment by pleblanc1976** [comment]
In Centos 6, we're getting this error:  "error: #pragma GCC diagnostic not allowed inside functions"
so, would need to wrap whole functions apparently, or test for compiler version and uglify the code.

Centos 7 and Deb 8 apparently don't like the format-truncation suppressions, may be a syntax thing though:  "warning: unknown option after ‘#pragma GCC diagnostic’ kind [-Wpragmas]
 #pragma GCC diagnostic ignored "-Wformat-truncation="

Yeah, we'll need to iterate on this a little to get it to compile.

**Review by pleblanc1976 — CHANGES_REQUESTED** [review]
Need to make compilers happy.

File: primitives/linux-port/column.cpp
Comment (drrtuy) [reply primitives/linux-port/column.cpp:1424]: All starting from gcc 7.

File: tools/dbloadxml/colxml.cpp
Comment (drrtuy) [reply tools/dbloadxml/colxml.cpp]: I agree with that guess. I will add extra message that notifies about the failure to simplify troubleshooting.

File: primitives/linux-port/column.cpp
Comment (drrtuy) [reply primitives/linux-port/column.cpp:1424]: pragma is another option how to disable warnings. But I prefer to have less pragmas disables the warnings throughout the code.

File: utils/threadpool/threadpool.h
Comment (drrtuy) [inline utils/threadpool/threadpool.h:81]: This could potentially break SUSE12 build.

Comment (drrtuy) [reply utils/threadpool/threadpool.h:81]: As I expected there is no std::unique_ptr in centos 6/suse 12 stdlib. I think I have to put preprocessor conditional expression to check for GCC version higher then 5.

**Comment by pleblanc1976** [comment]
Odd, looks like BB only ran the windows check.  Will check with Ben.

**Comment by pleblanc1976** [comment]
Ben says there are a lot of build jobs queued atm, and the ones for this PR are scheduled.  Will wait for results.

File: utils/threadpool/threadpool.h
Comment (drrtuy) [reply utils/threadpool/threadpool.h:81]: Couldn't do much about the warning produced in U16. g++ uses 97 standard by default but std::unique_ptr is supported since c++11.

Comment (pleblanc1976) [reply utils/threadpool/threadpool.h:81]: the Boost smart ptrs will work.  std::unique_ptr == boost::scoped_ptr

**Review by pleblanc1976 — APPROVED** [review]

**Comment by pleblanc1976** [comment]
BB is still failing in U16 & C7, but it looks like it has nothing to do with this.

I'll leave it to you to decide whether to switch ptr types to eliminate the warning in U16.

**Pull request merged**
